source/main.py

```diff
@@ -1,50 +1,62 @@
 import common.midi_processing as midi_processing
 import image_processing_ai as img_proc_ai
+import image_processing as img_proc
+import take_image
 import multiprocessing as mp
 
-cap = cv2.VideoCapture(0)
 img_proc_ai.setup_ai()
 
 import os
-
+import time
 if __name__ == "__main__":
-    img_path = os.path.join(os.getcwd(), 'files','image_processing','example_shapes (3).png')
-    img = cv2.imread(img_path)
-    assert img is not None, "file could not be read, check with os.path.exists()"
+    while True:
+    
+        # BUG: Currently i did not get any shapes working with the camera test
+        img = take_image.get_image()
+        # img_path = os.path.join(os.getcwd(), 'files','image_processing','example_shapes (3).png')
+        # img = cv2.imread(img_path)
+
+        image_ai, list_of_shapes = img_proc_ai.detect_shapes_with_ai(img)
+        cv2.destroyAllWindows()
+        cv2.imshow('AI processing', image_ai)
+        cv2.waitKey(1)# Displays the new image immediately
+        img_proc.display_list_of_shapes(list_of_shapes)
 
-    # while(1):
-        # success, img = cap.read()
-    image_ai, list_of_shapes = img_proc_ai.detect_shapes_with_ai(img)
+        bpm = midi_creation.MakeSong(list_of_shapes) 
+        
+        processes = [
+            mp.Process(target=midi_processing.instrument, args=(bpm, "drum")),
+            mp.Process(target=midi_processing.instrument, args=(bpm, "violin")),
+            mp.Process(target=midi_processing.instrument, args=(bpm, "guitar")),
+            mp.Process(target=midi_processing.instrument, args=(bpm, "flute")),
+            mp.Process(target=midi_processing.instrument, args=(bpm, "saxophone")),
+            mp.Process(target=midi_processing.instrument, args=(bpm, "clap")),
+            mp.Process(target=midi_processing.instrument, args=(bpm, "piano"))
+        ]
+        
+        # Start all processes
+        for process in processes:
+            process.start()
 
-    img_proc_ai.detect_shapes_with_ai(img)
-    img_proc.display_list_of_shapes(list_of_shapes)
+        # Wait for all processes to finish
+        for process in processes:
+            process.join()
+            
+        midi_processing.audio_rendering(bpm)
 
-    bpm = midi_creation.MakeSong(list_of_shapes) 
-    
-    processes = [
-        mp.Process(target=midi_processing.instrument, args=(bpm, "drum")),
-        mp.Process(target=midi_processing.instrument, args=(bpm, "violin")),
-        mp.Process(target=midi_processing.instrument, args=(bpm, "guitar")),
-        mp.Process(target=midi_processing.instrument, args=(bpm, "flute")),
-        mp.Process(target=midi_processing.instrument, args=(bpm, "saxophone")),
-        mp.Process(target=midi_processing.instrument, args=(bpm, "clap")),
-        mp.Process(target=midi_processing.instrument, args=(bpm, "piano"))
-    ]
-    
-    # Start all processes
-    for process in processes:
-        process.start()
+        cv2.destroyAllWindows()
```

Comment:
The titel of the window has the intructions in it.
We do destroy all windows to make sure the others are closed.
otherwise it would just open a new one.